sys/net/transport_layer/ng_udp/ng_udp.c

```diff
@@ -113,9 +113,10 @@ static void _receive(ng_pktsnip_t *pkt)
     }
     hdr = (ng_udp_hdr_t *)udp->data;
 
+    LL_SEARCH_SCALAR(pkt, ipv6, type, NG_NETTYPE_IPV6);
```

Comment:
I'd say there will be only one IPv6 header on receive. If there is 6in6 encapsulation the additional header will be removed in my current concept.